Properties config with partial “appenders” list fails in >= 2.25.0 with ConfigurationException: No type attribute provided for component appender

## Description
Upgrading to Log4j 2.25.0 introduces a regression in properties-based configuration when the top-level appenders property lists only a subset of the defined appenders. At startup, Log4j throws:
ConfigurationException: No type attribute provided for component appender

This appears to be caused by the new processing of “remaining” root-level properties in PropertiesConfigurationBuilder.build(), which now folds leftover dotted prefixes (like appender.) into synthetic components and then requires a type on the container key appender. In prior releases, extra appender. groups not listed in appenders did not cause a failure. 
It was introduced by [commit](https://github.com/apache/logging-log4j2/commit/c299479b03aa0eeb2baca64e39b017d649b0e09b#diff-a398802ec75cb65f7576c05a7235bb60947c2fa51d1ca57f70bc667f45597724)

## Expected behavior
- Properties configs that specify **appenders = …** should not fail if additional **appender.<id>.*** groups exist but are not listed.
1. Only the listed (or referenced) **appenders** should be instantiated; unlisted groups should not trigger a root-level component parse or an exception. 

## Configuration
**Version:** >= 2.25.0

```
# Root logger configuration
status = warn
name = PropertiesConfig

appenders = console

# Console appender
appender.console.type = Console
appender.console.name = Console
appender.console.target = SYSTEM_OUT
appender.console.layout.type = PatternLayout
appender.console.layout.pattern = %d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n

appender.console2.type = Console
appender.console2.name = Console
appender.console2.target = SYSTEM_OUT
appender.console2.layout.type = PatternLayout
appender.console2.layout.pattern = %d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n
```

## Logs

```
Caused by: org.apache.logging.log4j.core.config.ConfigurationException: No type attribute provided for component appender
	at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationBuilder.createComponent(PropertiesConfigurationBuilder.java:362)
	at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationBuilder.processRemainingProperties(PropertiesConfigurationBuilder.java:206)
	at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationBuilder.build(PropertiesConfigurationBuilder.java:190)
Caused by: org.apache.logging.log4j.core.config.ConfigurationException: No type attribute provided for component appender

	at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationFactory.getConfiguration(PropertiesConfigurationFactory.java:56)
	at org.apache.logging.log4j.core.config.properties.PropertiesConfigurationFactory.getConfiguration(PropertiesConfigurationFactory.java:34)
	at org.apache.logging.log4j.core.config.ConfigurationFactory$Factory.getConfiguration(ConfigurationFactory.java:544)
	at org.apache.logging.log4j.core.config.ConfigurationFactory$Factory.getConfiguration(ConfigurationFactory.java:463)
	at org.apache.logging.log4j.core.config.ConfigurationFactory.getConfiguration(ConfigurationFactory.java:321)
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:778)
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:808)
	at org.apache.logging.log4j.core.LoggerContext.start(LoggerContext.java:311)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:160)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:48)
	at org.apache.logging.log4j.LogManager.getContext(LogManager.java:139)
	at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:557)
	at com.l4jmre.App.<clinit>(App.java:11)
```

## Reproduction
An example configuration throws the above exception. 

## Workarounds
- Remove the appenders line entirely (let names be inferred), or
- List all defined appenders: appenders = console, console2 

Both avoid the synthetic root-level “component appender” and the resulting exception.

--- a/log4j-core-test/src/test/resources/log4j-rolling.properties
+++ b/log4j-core-test/src/test/resources/log4j-rolling.properties
@@ -51,6 +51,11 @@ appender.my.list.filters = threshold
 appender.my.list.filter.threshold.type = ThresholdFilter
 appender.my.list.filter.threshold.level = error
 
+appender.unused.type = RollingFile
+appender.unused.name = UnusedRollingFile
+appender.unused.fileName = ${filename}.unused
+appender.unused.filePattern = target/rolling2/test1-%d{MM-dd-yy-HH-mm-ss}-unused-%i.log.gz
+
 loggers = rolling.file
 
 logger.rolling.file.name = org.apache.logging.log4j.core.appender.rolling

--- a/log4j-core/src/main/java/org/apache/logging/log4j/core/config/properties/PropertiesConfigurationBuilder.java
+++ b/log4j-core/src/main/java/org/apache/logging/log4j/core/config/properties/PropertiesConfigurationBuilder.java
@@ -131,6 +131,7 @@ public class PropertiesConfigurationBuilder extends ConfigurationBuilderFactory
                 final String name = filterName.trim();
                 builder.add(createFilter(name, PropertiesUtil.extractSubset(rootProperties, "filter." + name)));
             }
+            removeDefinedButUnusedProperties("filter");
         } else {
 
             final Map<String, Properties> filters =
@@ -148,6 +149,7 @@ public class PropertiesConfigurationBuilder extends ConfigurationBuilderFactory
                 builder.add(createAppender(
                         appenderName.trim(), PropertiesUtil.extractSubset(rootProperties, "appender." + name)));
             }
+            removeDefinedButUnusedProperties(Appender.ELEMENT_TYPE);
         } else {
             final Map<String, Properties> appenders = PropertiesUtil.partitionOnCommonPrefixes(
                     PropertiesUtil.extractSubset(rootProperties, Appender.ELEMENT_TYPE));
@@ -165,6 +167,7 @@ public class PropertiesConfigurationBuilder extends ConfigurationBuilderFactory
                     builder.add(createLogger(name, PropertiesUtil.extractSubset(rootProperties, "logger." + name)));
                 }
             }
+            removeDefinedButUnusedProperties("logger");
         } else {
 
             final Map<String, Properties> loggers = PropertiesUtil.partitionOnCommonPrefixes(
@@ -192,6 +195,10 @@ public class PropertiesConfigurationBuilder extends ConfigurationBuilderFactory
         builder.setLoggerContext(loggerContext);
 
         return builder.build(false);
+    }
+
+    private void removeDefinedButUnusedProperties(final String prefix) {
+        PropertiesUtil.extractSubset(rootProperties, prefix);
     }
 
     private void processRemainingProperties(

--- a/src/changelog/.2.x.x/prevent_configexception_non_used_config.xml
+++ b/src/changelog/.2.x.x/prevent_configexception_non_used_config.xml
@@ -1,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<entry xmlns="https://logging.apache.org/xml/ns"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="
+           https://logging.apache.org/xml/ns
+           https://logging.apache.org/xml/ns/log4j-changelog-0.xsd"
+       type="added">
+    <issue id="4036" link="https://github.com/apache/logging-log4j2/issues/4036"/>
+    <issue id="4069" link="https://github.com/apache/logging-log4j2/pull/4069"/>
+    <description format="asciidoc">
+        Disregarded defined logger, appender and filter properties not referenced in loggers, appenders and filters respectively
+    </description>
+</entry>

